File: java/src/main/java/mx/iteso/Commenter.java
package mx.iteso;

/** Commenter privlege class. */
public class Commenter extends PrivilegesDecorator {
    /** Real User to decorate. */
    private final User user;

    /**
     * Commenter constructor.
     * @param incomingUser user.
     */
    public Commenter(final User incomingUser) {
        this.user = incomingUser;
    }

    /**
     * Description of permission values.
     * @return String description.
     */
    @Override
    public String getDescription() {
        return "Commenter: " + user.getDescription();
    }

    /**
     * Publish permission obtain function.
     * @return boolean Permission.
     */
    public boolean publishPermission() {
        return user.publishPermission();
    }

    /**
     * Deletion permission obtain function.
     * @return boolean Permission.
     */
    public boolean deletionPermission() {
        return user.deletionPermission();
    }

    /**
     * Approval permission obtain function.
     * @return boolean Permission.
     */
    public boolean approvalPermission() {
        return user.approvalPermission();
    }

    /**
     * Comment permission allow function.
     * @return boolean Permission.
     */
    public boolean commentPermission() {
        System.out.println("Commenting permission enabled");
        return true;
    }
}

File: python/src/main/python/Commenter.py

class Commenter():
    def __init__(self,incomingUser):
        self.user = incomingUser

    def getDescription(self):
        return "Commenter: " + self.user.getDescription()

    def publishPermission(self):
        return self.user.publishPermission()

    def deletionPermission(self):
        return self.user.deletionPermission()

    def approvalPermission(self):
        return self.user.approvalPermission()


    def commentPermission(self):
        print("Commenting permission enabled")
        return True
